WIP: Video Autocaptioner

zsxkib opened the pull request.

zsxkib committed:
basic autocaptioning works

zsxkib committed:
feat: improve logging and UX 🚀

zsxkib committed:
fix: enforce gradient checkpointing for 80GB VRAM training

daanelson commented on cog.yaml:
```diff
@@ -113,8 +113,18 @@ build:
-    - curl -o /usr/local/bin/pget -L "https://github.com/replicate/pget/releases/download/v0.8.1/pget_linux_x86_64" && chmod +x /usr/local/bin/pget
+    - FLASH_ATTENTION_SKIP_CUDA_BUILD=TRUE pip install flash-attn --no-build-isolation
+    - curl -o /usr/local/bin/pget -L "https://github.com/replicate/pget/releases/download/v0.8.2/pget_linux_x86_64" && chmod +x /usr/local/bin/pget
     - pip install onnxruntime-gpu --extra-index-url https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/onnxruntime-cuda-12/pypi/simple/
```
out of total curiosity - what here needs the onnx runtime?

zsxkib replied:
It was just part of comfyui afaik https://github.com/fofr/cog-comfyui/blob/main/cog.yaml

daanelson commented on train.py:
```diff
@@ -2,16 +2,59 @@
+
+# Configure logging to suppress INFO messages
+logging.basicConfig(level=logging.WARNING, format="%(message)s")
+
```
what's the point of suppressing all this logging?

zsxkib replied:
It's too verbose without these

daanelson commented on train.py:
```diff
@@ -278,15 +365,17 @@ def run_lora_training(
         OUTPUT_DIR,
         "--output_name",
         "lora",
```
are we hitting OOMs without gradient checkpointing?

zsxkib replied:
yeah we are ://

daanelson commented on train.py:
```diff
@@ -25,49 +68,60 @@ class TrainingOutput(BaseModel):
+    trigger_word: str = Input(
+        description="The trigger word refers to the object, style or concept you are training on. Pick a string that isn't a real word, like TOK or something related to what's being trained, like STYLE3D. The trigger word you specify here will be associated with all videos during training. Then when you use your LoRA, you can include the trigger word in prompts to help activate the LoRA.",
+        default="TOK",
+    ),
```
naive question - should we default to autocaptioning or default to not autocaptioning?

daanelson replied:
ah jk I see that we only autocaption videos w/out captions. makes more sense now. might want to call that behavior out.

daanelson commented on train.py:
```diff
@@ -335,41 +430,170 @@ def download_weights():
+                    clean_up_tokenization_spaces=False,
+                )[0]
+
+                # Combine prefix, trigger, caption, and suffix
```
very nit - do we want spaces between these? something like `" ".join([prefix, trigger, caption.strip(), suffix])`

daanelson commented on train.py:
```diff
@@ -335,41 +430,170 @@ def download_weights():
+            print(f"✅ Saved to: {txt_path}")
+
+    # Clean up QWEN model
+    print("\n=== 🧹 Cleaning Up ===")
```
would change to `Cleaning Up autocaptioner`

daanelson commented on train.py:
```diff
@@ -335,41 +430,170 @@ def download_weights():
+
+            try:
+                # Import qwen utils here to avoid circular imports
+                from qwen_vl_utils import process_vision_info
```
curious where the circular import comes from, probably fine though

daanelson commented on train.py:
```diff
@@ -467,3 +703,28 @@ def handle_hf_readme(hf_repo_id: str) -> str:
+
+def setup_qwen_model():
+    """Download and setup Qwen2-VL model for auto-captioning"""
+    # TODO: use download_weights() instead
```
might as well use `download_weights` before merging, should be quick.

daanelson reviewed (commented):
neat. generally looks good overall! added some comments here and there. 

interesting that the trainer requires cacheing latents outside of the main training process; I would think that's an inefficiency we could handle down the line but also probably pretty small in the context of the overall runtime.

zsxkib committed:
WIP: Add max_train_steps to control training duration

zsxkib committed:
add input validation for dimensions, frames, and enhancement params

- Ensures width/height are multiples of 16
- Validates frame count follows 4n+1 pattern
- Validates enhancement start/end are between 0-1
- Adds helpful warnings when values are adjusted

zsxkib committed:
Update README.md
